pkg/v5/resource/chart/current.go

```diff
@@ -40,7 +40,15 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 	if helmclient.IsReleaseNotFound(err) {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the chart-operator chart in the guest cluster")
 		return nil, nil
+	} else if guest.IsAPINotAvailable(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "guest cluster is not available")
+
+		// We can't continue without a successful K8s connection. Cluster
+		// may not be up yet. We will retry during the next execution.
+		reconciliationcanceledcontext.SetCanceled(ctx)
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation for custom object")
```

Comment:
BTW can you align with what we discussed in `fmt`?

Reply:
Sure will do. Although it isn't merged yet ;)

Reply:
I think it should be

```
// did not find
r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the chart-operator chart in the guest cluster")
// reason
r.logger.LogCtx(ctx, "level", "debug", "message", "guest cluster is not available")
```

Reply:
Ack but I only made the logging changes to `v6`. For `v5` I just backported the fix. Going with this for now. I'll get the logging in `v6` straight in a later PR.